merlion/models/automl/autoets.py

```diff
@@ -30,3 +92,166 @@ class AutoETS(SeasonalityLayer):
     """
 
     config_class = AutoETSConfig
+
+    def generate_theta(self, train_data: TimeSeries) -> Iterator:
+        """
+        generate [theta]. theta is a list of parameter combination [error, trend, damped_trend, seasonal]
+        """
+        y = train_data.univariates[self.target_name].np_values
+
+        # check the size of y
+        n_samples = y.shape[0]
+        if n_samples <= 3:
+            self.information_criterion = "aic"
+
+        # auto-detect seasonality if desired, otherwise just get it from seasonal order
+        if self.config.auto_seasonality:
+            candidate_m = super().generate_theta(train_data=train_data)
+            self.config.seasonal_periods, _, _ = super().evaluate_theta(thetas=candidate_m, train_data=train_data)
```

Comment:
Logically, it doesn't make sense to me for seasonality to be set in `generate_theta()`. This value should be returned by this method and handled in `set_theta()`. If `self.config.auto_seasonality` is `False`, then we should return `self.model.seasonal_periods` instead.

Reply:
Here I follow the pipeline of `AutoSarima`, `generate_theta()` will handle periodicity detection and extra hyper-parameters grid search.  The logic is once we finish all the automl procedures and find the best model, then we create a new ETS model with the config file found by AutoETS.